StatBot/EchoBot.cs

```diff
@@ -23,49 +36,206 @@ public class EchoBot : IBot
+
+
+                        var baseInterfaceType = typeof(ITool);
+                        var botCommands = Assembly.GetAssembly(baseInterfaceType)
+                            .GetTypes()
+                            .Where(types => types.IsClass && !types.IsAbstract && types.GetInterface("ITool") != null);
+                        foreach (var botCommand in botCommands)
+                        {
+                            _tools.Add((ITool)Activator.CreateInstance(botCommand));
+                        }
+
+                        var str = context.Activity.Text.Trim();
+                        var indexOfSpace = str.IndexOf(" ", StringComparison.Ordinal);
+                        var command = indexOfSpace != -1 ? str.Substring(0, indexOfSpace).ToLower() : str.ToLower();
+                        if (command[0] != '/')
                         {
-                            await context.SendActivity(help.Run(context.Activity));
-                            return;
+                            command = "/" + command;
                         }
-                        await context.SendActivity(tool.Run(context.Activity));
+
+                        var help = new Help();
+
+                        var tool = _tools.FirstOrDefault(x => x.CommandsName.Any(y => y.Equals(command)));
+                        if (tool != null)
+                        {
+                            context.Activity.Text = indexOfSpace >= 0 ? context.Activity.Text.Substring(indexOfSpace + 1, str.Length - indexOfSpace - 1) : String.Empty;
+                            if (user == null || (!user.IsAdmin && ((ITool)tool).IsAdmin))
+                            {
+                                await context.SendActivity(help.Run(context.Activity));
+                                return;
+                            }
+                            await context.SendActivity(tool.Run(context.Activity));
+                        }
+                        //else
+                        //{
+                        //    await context.SendActivity(help.Run(context.Activity));
+                        //}
+                    }
+                }
+            }
+        }
+
+        public void InitDialog()
+        {
+            dialogs.Add("createStatDialig", new WaterfallStep[]
+            {
+                async (dc, args, next) =>
+                {
+                    dc.ActiveDialog.State = new Dictionary<string, object>();
+                    // Prompt for the guest's name.
+
+                    await dc.Prompt("textPrompt1", "Введите имя статистики");
+                },
+                async(dc, args, next) =>
+                {
+                    dc.ActiveDialog.State["name"] = args["Value"];
+                    var userState = UserState<BotUserState>.Get(dc.Context);
+                    userState.statName = Convert.ToString(dc.ActiveDialog.State["name"]);
+
+                    //reservationDate = Convert.ToDateTime(dateTimeResult.Value);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://172.21.13.172:9091/coding_rules#rule_key=csharpsquid%3AS125)